uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/WorkPatternCalendar.php

```diff
@@ -314,4 +293,152 @@ private function calculateWorkPatternPeriodEndDate(ContactWorkPattern $contactWo
     return $patternEffectiveEndDate;
   }
 
+  /**
+   * This method fills the dates lapses between the items in the array passed in using
+   * the fabricateMethod.
+   * It also fills lapses between the absence period
+   * start date and the first item in the array and the also between the last item
+   * and the absence period end date using the fabricate method provided.
+   *
+   * The array passed in could be Work Pattern periods
+   * or Contracts.
+   *
+   * @param array $toFabricateFor
+   * @param string $fabricateMethod
+   *
+   * @return array
+   *   An array with the no date lapses in between
+   */
+  private function fabricateForLapses($toFabricateFor, $fabricateMethod) {
+    $fabricated = [];
```

Comment:
I agree its a generic function and can be part of a separate class. But like you said I'll leave it for now since its probably only the calendar class that makes use of it presently.